File: 2006-find-the-student-that-will-replace-the-chalk/2006-find-the-student-that-will-replace-the-chalk.java
class Solution {
    public int chalkReplacer(int[] chalks, long k) {
        long totalChalks = 0;
        for (int chalk: chalks) {
            totalChalks += chalk;
        }

        k = (int) k % totalChalks;

        int replace = 0;

        for (int i = 0; i < chalks.length; i++) {
            if (chalks[i] > k) {
                replace = i;
                break;
            } else {
                k -= chalks[i];
            }
        }

        return replace;
    }
}

File: 2006-find-the-student-that-will-replace-the-chalk/2006-find-the-student-that-will-replace-the-chalk.go
func chalkReplacer(chalks []int, k int) int {
    totalChalks := 0
    for _, chalk := range chalks {
        totalChalks = totalChalks + chalk
    }

    k = k % totalChalks

    replace := 0

    for i, chalk := range chalks {
        if chalk > k {
            replace = i
            break
        }

        k = k - chalk
    }

    return replace
}